lib/ansible/executor/process/worker.py

```diff
@@ -77,7 +79,10 @@ def __init__(self, rslt_q, task_vars, host, task, play_context, loader, variable
                 fileno = sys.stdin.fileno()
                 if fileno is not None:
                     try:
-                        self._new_stdin = os.fdopen(os.dup(fileno))
+                        if PY3: # needed for recognizing Enter in 'pause' module
+                            self._new_stdin = os.fdopen(os.dup(fileno), newline='\r\n')
```

Comment:
To fix #26278 the intention is to deactivate "universal newlines mode" to exit the prompt correctly.

- when **newline=None** (default): Prompt is never exiting, ctrl+c is needed to interrupt
- when **newline=""**: prompt only exiting when pressing Enter and again any key at the keyboard
- when **newline is \n, \r or \r\n**: prompt is exiting as expected

First case is clear imho, all newline like symbols are translated to \n and stdin will never exit.
Second case is more tricky as I don't have that deep knowledge about python core functions. Maybe "universal newlines mode" is buffering single newline chars somehow and messing up the stdin?